spec/system/admin/reports/orders_and_fulfillment_spec.rb

```diff
@@ -95,6 +95,14 @@
                             ])
       end
 
+      it "pre selects the last order cycle when it exists" do
+        order_cycle = create(:simple_order_cycle, distributors: [distributor])
```

Comment:
Thank you @binarygit !

I think we could as well define the `order_cycle` in the top of the file using `let` - this way we would not need to reload the page with `visit current_path`, which is expensive in terms of performance in system specs. However, I don't think this should block moving forward.

Moving to test ready :muscle: 

I'll propose these changes on a separate PR:


```
diff --git a/spec/system/admin/reports/orders_and_fulfillment_spec.rb b/spec/system/admin/reports/orders_and_fulfillment_spec.rb
index db39f9ab6..f2e022763 100644
--- a/spec/system/admin/reports/orders_and_fulfillment_spec.rb
+++ b/spec/system/admin/reports/orders_and_fulfillment_spec.rb
@@ -18,13 +18,14 @@ describe "Orders And Fulfillment" do
       create(:address, address1: "distributor address", city: 'The Shire', zipcode: "1234")
     }
     let(:distributor) { create(:distributor_enterprise, address: distributor_address) }
+    let(:order_cycle) { create(:simple_order_cycle, distributors: [distributor]) }
     let(:order1) {
       create(:completed_order_with_totals, line_items_count: 0, distributor: distributor,
-                                           bill_address: bill_address1)
+                                           bill_address: bill_address1, order_cycle_id: order_cycle.id)
     }
     let(:order2) {
       create(:completed_order_with_totals, line_items_count: 0, distributor: distributor,
-                                           bill_address: bill_address1)
+                                           bill_address: bill_address1, order_cycle_id: order_cycle.id)
     }
     let(:supplier) { create(:supplier_enterprise, name: "Supplier") }
     let(:product) { create(:simple_product, name: "Product", supplier: supplier ) }
@@ -95,6 +96,10 @@ describe "Orders And Fulfillment" do
                             ])
       end
 
+      it "pre selects the last order cycle when it exists" do
+        expect(find('#q_order_cycle_id_in').value).to eq [order_cycle.id.to_s]
+      end
+
       it "handles order cycles with nil opening or closing times" do
         distributor = create(:distributor_enterprise)
         oc = create(:simple_order_cycle, name: "My Order Cycle", distributors: [distributor],

```

Reply:
@filipefurtad0 Good idea! Just a couple of little things I would change:

* I would assign `order_cycle` instead of `order_cycle_id`. The advantage is that the created object is already loaded and assigned instead of just assigning an id which means that the order cycle record has to be fetched from the database again when it's accessed. Not a big difference here but in other specs it can be quite important to work with in-memory objects.
* I would put the order_cycle parameter on a new line for readability.

Another improvement would be to test that we are selected the *last* order cycle. I believe that we have only one after your spec change. The original spec creates an additional one. But it's very unlikely to get this wrong and this bug wouldn't be that bad if it happened. So I'm tempted to say: let's not bother and keep the spec simple unless we actually encounter a bug and then we write a spec for that.

Reply:
Thanks @mkllnk, for the suggestions and that great catch: I think the test case you mention is indeed relevant. I'll add/consider this on https://github.com/openfoodfoundation/openfoodnetwork/pull/9406 :+1